scripts/Engineering/EnginX.py

```diff
@@ -622,28 +622,41 @@ def _save_out(run_number, focus_directory, focus_general, output, join_string, b
 
     """
     # work out where to save the files
-    filename = os.path.join(focus_directory, join_string.format(run_number, bank_id))
-    hdf5_name = os.path.join(focus_directory, run_number + ".hdf5")
+    dat_name, genie_filename, gss_name, hdf5_name, nxs_name = _find_focus_file_location(bank_id, focus_directory,
+                                                                                       join_string, run_number)
     if not u(bank_id).isnumeric():
         bank_id = 0
     # save the files out to the user directory
-    simple.SaveFocusedXYE(InputWorkspace=output, Filename=filename + ".dat", SplitFiles=False,
+    simple.SaveFocusedXYE(InputWorkspace=output, Filename=dat_name, SplitFiles=False,
                           StartAtBankNumber=bank_id)
-    simple.SaveGSS(InputWorkspace=output, Filename=filename + ".gss", SplitFiles=False, Bank=bank_id)
-    simple.SaveOpenGenieAscii(InputWorkspace=output, Filename=filename + ".his", OpenGenieFormat="ENGIN-X Format")
-    simple.SaveNexus(InputWorkspace=output, Filename=filename + ".nxs")
+    simple.SaveGSS(InputWorkspace=output, Filename=gss_name, SplitFiles=False, Bank=bank_id)
+    simple.SaveOpenGenieAscii(InputWorkspace=output, Filename=genie_filename, OpenGenieFormat="ENGIN-X Format")
+    simple.SaveNexus(InputWorkspace=output, Filename=nxs_name)
     simple.ExportSampleLogsToHDF5(InputWorkspace=output, Filename=hdf5_name, Blacklist="bankid")
     if not focus_general == focus_directory:
         if not os.path.exists(focus_general):
             os.makedirs(focus_general)
         # copy the files to the general directory
-        copy2(filename + ".dat", focus_general)
-        copy2(filename + ".gss", focus_general)
-        copy2(filename + ".his", focus_general)
-        copy2(filename + ".nxs", focus_general)
+        copy2(dat_name, focus_general)
+        copy2(gss_name, focus_general)
+        copy2(genie_filename, focus_general)
+        copy2(nxs_name, focus_general)
         copy2(hdf5_name, focus_general)
 
 
+def _find_focus_file_location(bank_id, focus_directory, join_string, run_number):
+
+    run_and_bank = join_string.format(run_number, bank_id, "{}")
```

Comment:
I think a comment to say that `{}` is replaced with `{}` so that different extensions can be added later might be nice, as it's not immediately clear. What do you think?